sql/derror.cc

```diff
@@ -237,44 +237,10 @@ bool MY_LOCALE_ERRMSGS::read_texts() {
   if ((file = mysql_file_open(key_file_ERRMSG,
                               fn_format(name, ERRMSG_FILE, lang_path, "", 4),
                               O_RDONLY, MYF(0))) < 0) {
-<<<<<<< HEAD
-    /*
-      Trying pre-5.5 semantics of the --language parameter.
-      It included the language-specific part, e.g.:
-
-      --language=/path/to/english/
-    */
-    if ((file = mysql_file_open(
-             key_file_ERRMSG,
-             fn_format(name, ERRMSG_FILE, lc_messages_dir, "", 4), O_RDONLY,
-             MYF(0))) < 0) {
 #ifndef XTRABACKUP
```

Comment:
Can you please check what happens if u remove this ifndef? is pxb affected? do we get this ER_ERRMSG_CANT_FIND_FILE printed?

Reply:
thanks, next time I would prefer a more detailed reply. the stack trace is
(gdb) bt
#0  MY_LOCALE_ERRMSGS::read_texts (this=0x5555610ce680 <global_errmsgs>) at /home/satya/WORK/pxb-8.4/sql/derror.cc:241
#1  0x000055555911f0a8 in setup_error_messages () at /home/satya/WORK/pxb-8.4/storage/innobase/xtrabackup/src/xtrabackup.cc:7364
#2  0x0000555559144659 in main (argc=4, argv=0x7fffffffde18) at /home/satya/WORK/pxb-8.4/storage/innobase/xtrabackup/src/xtrabackup.cc:7415

and the error message printed is:
2024-07-26T14:14:07.764069+01:00 0 [ERROR] [MY-010338] [Server] Can't find error-message file 'english/errmsg.sys'. Check error-message file location and 'lc-messages-dir' configuration directive.

closing this conversation as marked